src/chat/chat_tests.rs

```diff
@@ -2964,15 +2964,23 @@ async fn test_sync_accept_before_first_msg() -> Result<()> {
 
     let ba_chat = bob.create_chat(alice0).await;
     let sent_msg = bob.send_text(ba_chat.id, "hi").await;
-    let a0b_chat_id = alice0.recv_msg(&sent_msg).await.chat_id;
-    assert_eq!(alice0.get_chat(&bob).await.blocked, Blocked::Request);
+    let rcvd_msg = alice0.recv_msg(&sent_msg).await;
+    let a0b_chat_id = rcvd_msg.chat_id;
+    let a0b_contact_id = rcvd_msg.from_id;
+    assert_eq!(
+        Chat::load_from_db(alice0, a0b_chat_id).await?.blocked,
+        Blocked::Request
+    );
     a0b_chat_id.accept(alice0).await?;
-    let a0b_contact = alice0.add_or_lookup_contact(&bob).await;
+    let a0b_contact = Contact::get_by_id(alice0, a0b_contact_id).await?;
```

Comment:
Just an idea: If we need to do this often, it might be worth it to create functions `sender_chat()` and `sender_contact()` on `ReceivedMsg` (defined in test_utils.rs so that it's only available in tests). If it's needed very often, could even make a new struct ReceivedMsgTest that dereferences to ReceivedMsg but contains a reference to the correct TestContext, so that you don't need to pass (and possibly pass the wrong TestContext) `&alice` to `sender_chat()` and `sender_contact()`.

Reply:
`sender_chat()` may be unclear: if it's the 1:1 chat with the sender, it may be different from the received message chat. But `sender_contact()` indeed looks useful